src/Service/ChangeSetFactory.php

```diff
@@ -241,6 +241,20 @@ protected function getChangeSet($entity = NULL, $relatedEntity = null)
 				if (isset($uowEntiyChangeSet[$property->getName()])) {
 					$propertyChangeSet = $uowEntiyChangeSet[$property->getName()];
 
+					if ($columnAnnotation->type === 'json') {
+						$clearedOldData = $clearedNewData = [];
+
+						foreach ($propertyChangeSet[0] as $key => $value) {
+							if ($propertyChangeSet[1][$key] !== $value) {
+								$clearedOldData[$key] = $value;
+								$clearedNewData[$key] = $propertyChangeSet[1][$key];
+							}
+						}
+
+						$propertyChangeSet[0] = json_encode($clearedOldData, JSON_PRETTY_PRINT);
```

Comment:
tady bych asi nedaval JSON_PRETTY_PRINT... ale hlavne pouzij Nette Json::encode, prootze normalni json ti zaenkoduje diakritiku (unicode znaky) a to nechceme...